core/activity/src/provider/service.rs

```diff
@@ -350,31 +406,41 @@ async fn monitor_activity(
                 break;
             }
 
-            let dt = (Utc::now().timestamp() - usage.timestamp) as f64;
-            if dt > limit_s {
-                log::warn!("activity {} inactive for {}s, destroying", activity_id, dt);
-                enqueue_destroy_evt(
-                    db,
-                    tracker.clone(),
-                    &activity_id,
-                    provider_id,
-                    app_session_id.clone(),
-                )
-                .await;
+            let duration = (Utc::now().timestamp() - usage.timestamp) as f64;
```

Comment:
Maybe?
```suggestion
            let since_update = (Utc::now().timestamp() - usage.timestamp) as f64;
```

Reply:
Terrible naming. I assumed it is the Activity duration.

Reply:
Changed.